libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/location/LocationPermissionComponent.kt

```diff
@@ -54,41 +52,10 @@ internal class LocationPermissionComponent(
     override fun onAttached(mapboxNavigation: MapboxNavigation) {
         super.onAttached(mapboxNavigation)
 
-        val isGranted = PermissionsManager.areLocationPermissionsGranted(
-            mapboxNavigation.navigationOptions.applicationContext
-        )
-
-        if (isGranted) {
-            // There can be a race condition between components and view models.
-            // The view model attaches, and then launches a coroutine to collect actions.
-            // The LocationPermissionComponent surfaces this issue because it is not owned by
-            // a coordinator and flowable binder. This issue was also difficult to reproduce on
-            // all devices. Launching a coroutine to update the state is a temporary solution.
```

Comment:
Although, it's not clear why it should be part of this PR.. IMO this can be an entirely separate change, it does not seem related to the "Reset SharedApp state when MapboxNavigation is destroyed"

Reply:
If you would see, I am allowing for location permissions `Only this time`. So, when I start the app again, why is it not asking me for location permissions? It seems like I already have the permission.

https://user-images.githubusercontent.com/9770186/178817858-ae5b059d-2cfd-45ce-9f80-b9073636a191.mp4

How do we handle the use case where I switch off my location and Drop-In UI doesn't works

https://user-images.githubusercontent.com/9770186/178817955-362ddd0f-a3b4-4e86-ac72-74574493d10b.mp4

@tomaszrybakiewicz

Reply:
Reverted changes to this file. Resolving this conversation.